.github/workflows/test.yml

```diff
@@ -31,27 +31,30 @@ jobs:
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
-      # Add or replace dependency steps here
-      - name: Install Chrome for integration/e2e tests
-        uses: nanasess/setup-chromedriver@v2.0.0
-      # The ruby version is taken from the .ruby-version file, no need to specify here.
-      - name: Install Ruby and gems
-        uses: ruby/setup-ruby@250fcd6a742febb1123a77a841497ccaa8b9e939
-        with:
-          bundler-cache: true
-      - name: Install frontend dependencies
-        run: npm ci
-      - name: Run Frontend build
-        run: bin/vite build
-      # Add or replace database setup steps here
-      - name: Create db
-        run: bin/rails db:migrate
-      - name: Set up database schema
-        run: bin/rails db:schema:load
+      - name: Setup Rails
+        uses: ./.github/workflows/build
       # Add or replace test runners here
-      - name: Run tests
+      - name: Run Ruby tests
         run: bundle exec rspec
-      # Add or replace any other lints here
+  lint:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
```

Comment:
Its a bit confusing that we need postgres to run the lint but this only seems necessary because the common build task runs db:migrate (along with installing chrome which also doesn't seem necessary to lint). Do we need to call the common build task for linting besides the need to install ruby/gems which might be simpler to just call directly within the linting job?

Reply:
I've gone with keeping the shared action and moving the postgres bit into the `build_and_test` action. The shared setup is now just installing Ruby and the BE and FE dependencies, but I still think that makes sense as a reusable bit of shared setup